mmpose/datasets/pipelines/mesh_transform.py

```diff
@@ -0,0 +1,370 @@
+
+        results['img'] = img
+        return results
+
+
+@PIPELINES.register_module()
+class MeshRandomFlip():
+    """Data augmentation with random image flip.
+
+    Required keys: 'img', 'joints_2d','joints_2d_visible', 'joints_3d',
+    'joints_3d_visible', 'center', 'pose', 'iuv' and 'ann_info'.
+    Modifies key: 'img', 'joints_2d','joints_2d_visible', 'joints_3d',
+    'joints_3d_visible', 'center', 'pose', 'iuv'.
+
+    Args:
+        flip_prob (float): Probability of flip.
+    """
+
+    def __init__(self, flip_prob=0.5):
+        self.flip_prob = flip_prob
+
+    def __call__(self, results):
+        """Perform data augmentation with random image flip."""
+        img = results['img']
+        joints_2d = results['joints_2d']
+        joints_2d_visible = results['joints_2d_visible']
+        joints_3d = results['joints_3d']
+        joints_3d_visible = results['joints_3d_visible']
+        pose = results['pose']
+        center = results['center']
+
+        if np.random.rand() <= self.flip_prob:
+            img = img[:, ::-1, :]
+            pose = flip_smpl_pose(pose)
+
+            joints_2d, joints_2d_visible = fliplr_joints(
+                joints_2d, joints_2d_visible, img.shape[1],
+                results['ann_info']['flip_pairs'])
+
+            joints_3d, joints_3d_visible = flip_joints_3d(
+                joints_3d, joints_3d_visible,
+                results['ann_info']['flip_pairs'])
+            center[0] = img.shape[1] - center[0] - 1
+
+        results['img'] = img
+        results['joints_2d'] = joints_2d
+        results['joints_2d_visible'] = joints_2d_visible
+        results['joints_3d'] = joints_3d
+        results['joints_3d_visible'] = joints_3d_visible
+        results['pose'] = pose
+        results['center'] = center
+
+        if 'iuv' in results.keys():
+            iuv = results['iuv']
+            if np.random.rand() <= self.flip_prob and iuv is not None:
+                iuv = flip_iuv(iuv, results['ann_info']['uv_type'])
+            results['iuv'] = iuv
+
+        return results
+
+
+@PIPELINES.register_module()
+class MeshGetRandomScaleRotation():
+    """Data augmentation with random scaling & rotating.
+
+    Required key: 'scale'. Modifies key: 'scale' and 'rotation'.
+
+    Args:
+        rot_factor (int): Rotating to ``[-2*rot_factor, 2*rot_factor]``.
+        scale_factor (float): Scaling to ``[1-scale_factor, 1+scale_factor]``.
+        rot_prob (float): Probability of random rotation.
+    """
+
+    def __init__(self, rot_factor=30, scale_factor=0.25, rot_prob=0.6):
+        self.rot_factor = rot_factor
+        self.scale_factor = scale_factor
+        self.rot_prob = rot_prob
+
+    def __call__(self, results):
+        """Perform data augmentation with random scaling & rotating."""
+        s = results['scale']
+
+        sf = self.scale_factor
+        rf = self.rot_factor
+
+        s_factor = np.clip(np.random.randn() * sf + 1, 1 - sf, 1 + sf)
```

Comment:
if you clip on `(1 - sf, 1 + sf)`, then 30% of the case, it will reach the two extreme values `1 - sf` and `1 + sf`. and only less than 70% of the values are in the range of `(1 - sf, 1 + sf)`. double check that this is intended.

Reply:
OK